src/action.ts

```diff
@@ -58,6 +59,15 @@ export const runMeticulousTestsAction = async (): Promise<void> => {
 
   const { base, head } = await getBaseAndHeadCommitShas(event);
   const environment = getEnvironment({ event, head });
+
+  await ensureBaseTestsExists({
+    event,
+    apiToken,
+    base,
+    context,
+    octokit,
+  });
```

Comment:
I wonder if we should later update this to:

```suggestion
  const { base: originalBase, head } = await getBaseAndHeadCommitShas(event);
  const environment = getEnvironment({ event, head });

  const base = await ensureBaseTestsExists({
    event,
    apiToken,
    base: originalBase,
    context,
    octokit,
  });
```

So that if we can only trigger a replay on the head of the main branch then we use the head of the main branch as our comparison, rather than an older commit. This could happen in the case that the main branch has updated since workflow started, or in the case that there's a merge conflict, so originalBase !== head commit of main. Currently it'd just run without performing any comparisons, but maybe it's better for it to compare to head of main in these cases, even if that means some spurious diffs?

Reply:
:thinking: I I think the API returns the `head_sha` of a workflow run, so it would be possible to get it and compare to `originalBase`.

Reply:
Yeah the other option would be to fail with an error if it differs, and ask the user to merge `baseRef` back into their branch.